src/clr-angular/utils/drag-and-drop/providers/drag-event-listener.service.ts

```diff
@@ -136,15 +151,36 @@ export class DragEventListenerService<T> {
             this.broadcast(endEvent, DragEventType.DRAG_END);
           }
 
+          clearTimeout(this.timeout);
+
           // We must remove the the nested listeners every time drag completes.
-          if (this.nestedListeners) {
-            this.nestedListeners.map(event => event());
-          }
+          this.nestedListeners.map(event => event());
+
+          // We must remove the event listener from checkDragStartBoundary
+          this.checkDragStartBoundaryListener();
```

Comment:
It seems unlikely, but it would be good to also check if this is defined and call it OnDestroy just to be safe.